lib/spack/spack/compilers/__init__.py

```diff
@@ -125,6 +125,105 @@ def get_compiler_config(scope=None, init_config=True):
     return config
 
 
+def get_compiler_config_from_packages(scope=None):
+    """Return the compiler configuration from packages.yaml"""
+    packages = []
+    config = spack.config.get("packages", scope=scope)
+
+    if not config:
+        return packages
+
+    for name, entry in config.items():
+        if name not in supported_compilers() and name not in package_name_to_compiler_name.keys():
+            continue
+        packages.extend(_compiler_config_from_package_config(entry))
+
+    return packages
+
+
+def _compiler_config_from_package_config(config):
+    externals = config.get("externals", None)
+    if not externals:
+        return []
+
+    compilers = []
+    for entry in externals:
+        compiler = _compiler_config_from_external(entry)
+        if compiler:
+            compilers.append(compiler)
+
+    return compilers
+
+
+def _compiler_config_from_external(config):
+    spec = spack.spec.Spec(config["spec"])
+    compiler_spec = spack.spec.CompilerSpec(
+        package_name_to_compiler_name.get(spec.name, spec.name), spec.version
+    )
+
+    extra_attributes = config.get("extra_attributes", {})
+    prefix = config.get("prefix", None)
+
+    compiler_class = class_for_compiler_name(compiler_spec.name)
+    paths = extra_attributes.get("paths", {})
+    compiler_langs = ["cc", "cxx", "fc", "f77"]
+    for compiler in compiler_langs:
+        if compiler in paths:
```

Comment:
We can set the default here, in case:
```suggestion
        if paths.setdefault(compiler, None):
```
...